gocql_bench/bench_marshal_test.go

```diff
@@ -0,0 +1,331 @@
+package gocql_bench
+
+import (
+	"testing"
+	"time"
+
+	"github.com/gocql/gocql"
+)
+
+// Benchmark marshalling and unmarshalling simple types
+func BenchmarkMarshalUnmarshalSimpleTypes(b *testing.B) {
+	b.Run("Int", func(b *testing.B) {
+		tType := gocql.NewNativeType(4, gocql.TypeInt, "")
+
+		var val int = 42
+		for i := 0; i < b.N; i++ {
+			// Marshal
+			marshaled, err := gocql.Marshal(tType, val)
+			if err != nil {
+				b.Fatal(err)
+			}
+
+			// Unmarshal
+			var unmarshaled int
+			err = gocql.Unmarshal(tType, marshaled, &unmarshaled)
+			if err != nil {
+				b.Fatal(err)
+			}
```

Comment:
Sorry for being confusing, but it is better to test Marshal and Unmarshal separately`

Reply:
but to test Unmarshal I need some marshled data, so in the test for unmarshaling I need to run both `Marshal()` and `Unmarshal()`? Or should we for now just test `Marshal`?

Reply:
call marshal once outside of the test just before t.run